### PR TITLE
Exams by Daria Rabiichuk

### PR DESCRIPTION
**Pull request by darr4i**
count.js and zip.js tasks

### CODE REVIEW

File: Tasks/zip.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
 // Zip two arrays, [1, 2] and [3, 4] -> [[1, 3], [2, 4]]
-'use strict'
+'use strict';
 
 const zip = (arr1, arr2) => {
   let i = 0, 
       j = 0;
   for (const element of arr2) {
     const tempCell = [arr1[i++], element];
     i < j ? delete arr1[i++] : arr2[j++] = tempCell;
```
Comment (RostislavHundarev) [inline Tasks/zip.js:9]: use assignments in your ternar operator

File: Tasks/zip.js [hunk 2]
```diff
@@ -1,18 +1,15 @@
...
-    CELL = [a[i++], x];
-    if (i < j) {
-      delete a[i++];
-    } else {
-      (() => (b[j++] = CELL))();
-    }
-    if (CELL[0] == undefined) b.length -= 1
+const zip = (arr1, arr2) => {
+  let i = 0, 
+      j = 0;
```
Comment (mari4kaa) [inline Tasks/zip.js:6]:
```suggestion
  let i = 0;
  let j = 0;
```

File: Tasks/zip.js [hunk 3]
```diff
@@ -1,18 +1,15 @@
...
+  for (const element of arr2) {
+    const tempCell = [arr1[i++], element];
+    i < j ? delete arr1[i++] : arr2[j++] = tempCell;
+    if (tempCell[0] == null) arr2.length -= 1;
```
Comment (mari4kaa) [inline Tasks/zip.js:10]: - Try not to change incoming parameters 
- Add the third "=": `if (tempCell[0] === null) arr2.length -= 1;`

**Review by mari4kaa — COMMENTED** [review]
I think that these changes will improve the code

File: Tasks/zip.js (hunk 1 above)
Comment (darr4i) [reply Tasks/zip.js]: Thanks! I simplified code

File: Tasks/zip.js (hunk 2 above)
Comment (darr4i) [reply Tasks/zip.js]: changed

File: Tasks/zip.js (hunk 3 above)
Comment (darr4i) [reply Tasks/zip.js]: Thank you! Now the arguments don't change and code works without conditions

**Comment by mari4kaa** [comment]
Excellent!!! I really like the way you changed the code in zip.js

File: Tasks/zip.js [hunk 4]
```diff
@@ -1,18 +1,13 @@
...
-    CELL = [a[i++], x];
-    if (i < j) {
-      delete a[i++];
-    } else {
-      (() => (b[j++] = CELL))();
-    }
-    if (CELL[0] == undefined) b.length -= 1
+const zip = (arr1, arr2) => {
+  const result = [];
+  for (let i = 0; i < arr1.length && i < arr2.length; i++) {
```
Comment (tshemsedinov) [inline Tasks/zip.js:6]: Alternative solution is to iterate over iterable returned from `entries()`
```suggestion
  for (const [i, value] of arr1.entries()) {
```

File: Tasks/count.js [hunk 5]
```diff
@@ -1,16 +1,12 @@
...
 
-count = (obj) => {
-  sum = 0;
-  {
-  }
-  keys = Object.keys(obj);
-  keys.forEach((key) => {
-    [];
-    const value = obj[key];
+const count = (obj) => {
```
Comment (tshemsedinov) [inline Tasks/count.js:4]: Naming hint: Better to rename `count` function to `sum` and rename `sum` variable to `result`

File: Tasks/count.js [hunk 6]
```diff
@@ -1,16 +1,12 @@
...
+  let sum = 0;
+  const values = Object.values(obj);
+  for(const value of values) {
```
Comment (tshemsedinov) [inline Tasks/count.js:7]:
```suggestion
  for (const value of values) {
```

File: Tasks/count.js [hunk 7]
```diff
@@ -1,16 +1,12 @@
...
     if (typeof value === 'number') sum += value;
```
Comment (tshemsedinov) [inline Tasks/count.js:8]: Alternative: use `reduce` to sum

**Review by tshemsedinov — APPROVED** [review]
LGTM (looks good to me)

File: Tasks/zip.js (hunk 4 above)
Comment (darr4i) [reply Tasks/zip.js]: added alternative solution

File: Tasks/count.js (hunk 5 above)
Comment (darr4i) [reply Tasks/count.js]: changed

**Pull request closed without merging**